sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -189,6 +189,16 @@ trait CheckAnalysis {
                s"but the left table has ${left.output.length} columns and the right has " +
                s"${right.output.length}")
 
+          case s: Union if s.children.exists(_.output.length != s.children.head.output.length) =>
+            s.children.filter(_.output.length != s.children.head.output.length).foreach { child =>
```

Comment:
I think we don't need `foreach` here, as the first exception will interrupt the analysis phase. We can just throw exception for the first error.